### PR TITLE
MINIFI-350 added pytest-based system integration test framework and initial test cases

### PR DESCRIPTION
**Pull request by achristianson**
Added a test framework which enables automated tests suits for the docker image as well as simulation/system integration testing of multiple-host flows and/or integration with other non-MiNiFi components.

Currently, the HTTP tests are failing due to https://issues.apache.org/jira/browse/MINIFI-369, but once the fix for that is merged, the test suite will pass.

### CODE REVIEW

**Comment by apiri** [comment]
reviewing

File: docker/test/integration/test_http.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+    listen_flow = ListenHTTP(8080) >> LogAttribute() >> PutFile('/tmp/output')
+
+    with DockerTestCluster(SingleFileOutputValidator('test')) as cluster:
+
+        cluster.put_test_data('test')
+        cluster.deploy_flow(listen_flow, name='minifi-listen')
+        cluster.deploy_flow(invoke_flow, name='minifi-invoke')
+
+        assert cluster.check_output()
```
Comment (apiri) [inline docker/test/integration/test_http.py]: This seems to be failing for me in OS X with Docker Mac.  Output below:

```
DEBUG:urllib3.connectionpool:http://localhost:None "DELETE /v1.30/networks/5f6cae20d1721e4ba01039fddabedc627532b5ec698897598cf6f18d2eed21d4 HTTP/1.1" 204 0
FAILED

==================================================================================== FAILURES ====================================================================================
_______________________________________________________________________________ test_invoke_listen _______________________________________________________________________________

    def test_invoke_listen():
        """
        Verify sending using InvokeHTTP to a receiver using ListenHTTP.
        """

        invoke_flow = (GetFile('/tmp/input') >> LogAttribute() >>
                       InvokeHTTP('http://minifi-listen:8080/contentListener', method='POST'))

        listen_flow = ListenHTTP(8080) >> LogAttribute() >> PutFile('/tmp/output')

        with DockerTestCluster(SingleFileOutputValidator('test')) as cluster:

            cluster.put_test_data('test')
            cluster.deploy_flow(listen_flow, name='minifi-listen')
            cluster.deploy_flow(invoke_flow, name='minifi-invoke')

>           assert cluster.check_output()
E           assert False
E            +  where False = <bound method DockerTestCluster.check_output of <minifi.test.DockerTestCluster object at 0x102a51e50>>()
E            +    where <bound method DockerTestCluster.check_output of <minifi.test.DockerTestCluster object at 0x102a51e50>> = <minifi.test.DockerTestCluster object at 0x102a51e50>.check_output

../docker/test/integration/test_http.py:36: AssertionError
====================================================================== 1 failed, 2 passed in 22.31 seconds =======================================================================
make[3]: *** [CMakeFiles/docker-verify] Error 1
make[2]: *** [CMakeFiles/docker-verify.dir/all] Error 2
make[1]: *** [CMakeFiles/docker-verify.dir/rule] Error 2
make: *** [docker-verify] Error 2
```

Comment (achristianson) [reply docker/test/integration/test_http.py]: The http tests are dependent on the phrocker/MINIFI-339 branch to pass, and possibly some of the other PRs I put in. There is unfortunately a bit of inter-dependence going on right now between the various branches.

**Comment by achristianson** [comment]
Thanks for the feedback. Adding more documentation is a good idea. I'll get on that and update the PR.

File: README.md [hunk 2]
```diff
@@ -124,6 +133,8 @@ $ brew install cmake \
   doxygen
 $ brew install curl
 $ brew link curl --force
+$ # (Optional) for building docker image/running system integration tests
+$ # Install docker using instructions at https://docs.docker.com/docker-for-mac/install/
```
[the remark this answers is not in the document]
Comment (achristianson) [reply README.md]: Fixed. Added instruction to pip install virtualenv.

**Comment by achristianson** [comment]
Added docs to docker/test/integration/README.md.

**Comment by apiri** [comment]
reviewing updates

File: docker/test/integration/.ropeproject/config.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
+# The default ``config.py``
```
Comment (apiri) [inline docker/test/integration/.ropeproject/config.py]: Does this file make sense to include?  Not familiar with rope.  If it has purpose, it does need a license header.

**Comment by apiri** [comment]
Hey @achristianson, 

Everything looks good.  Was unsure about the rope file that was created.  Let me know how we should handle that.

**Comment by achristianson** [comment]
The rope files should not have been committed at all.

**Comment by apiri** [comment]
Okay, I will remove those items and merge.  Thanks!

**Pull request closed without merging**